src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -240,15 +242,36 @@ def listDatasetFiles(self, datasetPath):
         Get list of files for dataset
 
         """
-        return [ x['logical_file_name'] for x in self.dbs.listFileArray(dataset = datasetPath)]
+        return [x['logical_file_name'] for x in self.dbs.listFileArray(dataset = datasetPath)]
 
-    def listDatatiers(self):
+    @staticmethod
+    def listDatatiers(dbsUrl=None):
         """
         _listDatatiers_
 
         Get a list of datatiers known by DBS.
         """
-        return [ tier['data_tier_name'] for tier in self.dbs.listDataTiers() ]
+        timenow = int(time.time())
+        if DBS3Reader._datatiers and timenow - 7200 < DBS3Reader._datatiers['ts']:
+            return DBS3Reader._datatiers['tiers']
+
+        DBS3Reader._setDatatiersCache(timenow, dbsUrl)
```

Comment:
may be we should try catch here, so if  cache updates fails return what is in the cache unless it is empty.
Also dbsUrl if mandatory paramater. it should check here not in _setDatatiersCache